node/podcontroller.go

```diff
@@ -300,7 +313,16 @@ func (pc *PodController) Run(ctx context.Context, podSyncWorkers int) (retErr er
 				pc.deletionQ.Forget(key)
 			}
 		},
-	})
+	}
+
+	if pc.eventFilterFunc != nil {
+		eventHandler = cache.FilteringResourceEventHandler{
```

Comment:
One thing that's neat about how this works in K8s, is if the filter function changes (for example, it says after 8pm, pods can start) the thing will re-sync. 

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/config/apiserver.go#L37-L48

Requests:
1. Can we ask the filter function to have a context (passed to it)
2. Can we maybe wrap the filter function and only allow pods through or some such?

Reply:
1. How does the context get passed to the filter?
2. I was thinking something like a function that bundles them together... so like `Filter(filters ...FilterFunc) FilterFunc`

Reply:
Added wrapper